src/control/build/interop.go

```diff
@@ -103,6 +114,25 @@ var defaultRules = []InteropRule{
 	},
 }
 
+var (
+	Server22xAgent20x = &InteropRule{
+		Self:          ComponentServer,
+		Other:         ComponentAgent,
+		Description:   "server v2.2.x is compatible with agent v2.0.x",
+		StopOnSuccess: true,
+		Check: func(self, other *VersionedComponent) bool {
+			// We assume that an unversioned agent is 2.0.x.
+			v0_0_0 := MustNewVersion("0.0.0")
+			v2_2_0 := MustNewVersion("2.1.100")
+			v2_3_0 := MustNewVersion("2.3.0")
+			return (self.Version.GreaterThanOrEquals(v2_2_0) &&
+				self.Version.LessThan(v2_3_0)) &&
+				(other.Version.GreaterThanOrEquals(v0_0_0) &&
```

Comment:
So we have a previous check to not support agent < v2.0, in that check we verify other is not < v2.0. Here we are assuming an unversioned agent (0.0.0) is v2.0 (presumably because v2.0 doesn't have the code to report version). In the first mentioned check, won't that fail because v2.0 reports 0.0.0 as per the assumption made here?

Reply:
Again, custom rules are processed before general rules, in order to allow for release-specific overrides. As the comment indicates on that general rule, it's a fallback in case no other rules supersede it.